### ISSUE TITLE
Tuned (LoRA) chat: CE-trained adapters are numeric-correct but not generation-capable (loradec emits NO_JSON_CONTRACT)

### ISSUE THREAD

**Issue by rickenator** [rickenator/VybForge#13]
## Status: infrastructure DONE + verified; objective gap remains
The tuned-slice runtime for :8888 is now fully built and numerically VERIFIED:
- kvresp-train gate PASS (corr 1.0 grads, CE loss 9.34->5.99 matching oracle): 89da0d8 (bulk DMA + deq_cached orientation/inz fix across all drivers).
- Trained adapters persisted (504 kvresp_l*.bin), validated.
- loradec_driver (LoRA autoregressive decode on GPU) runs clean, loads the fresh trained adapters + 9-token goal KV context, generates 49 tokens.

## The gap
Teacher-forced CE-trained adapters are numeric-correct but diverge at autoregressive generation:
LORADEC_VERIFY: NO_JSON_CONTRACT — decoded text is incoherent token-salad, no schema-valid agent-response JSON. Matches the exact caveat predicted in native/train/E2E-DECODE-MILESTONE.md.

## Fix needed (the remaining slice)
A short NEXT-TOKEN / RL-style fine-tune on the decode loop so the adapters learn to *emit* the {kind,message,...} contract, then re-run loradec + verify_loradec.py. After adapters generate a valid contract, wire the tuned path into native/host/chat_server.vyb (currently untuned-only, which works and answers correctly).

## Do NOT wire current adapters into chat
They'd return token-salad, strictly worse than the working untuned path. Chat stays untuned until the fine-tune lands.

## Repro
make -f native/Makefile kvresp-train-kv (adapters) -> kvctx_ctx_ids.bin (9-token goal ctx) -> loradec -> loradec_emit -> verify_loradec.py

**Comment by rickenator** [comment]
## Updated diagnosis (root cause actually found)
Fixing #13 turned out to be a DECODE bug, not (only) a training limitation:
- loradec_driver's deq_w dequantized every weight with hardcoded inz=0 (no orientation transpose) -> the autoregressive decode produced token-salad REGARDLESS of adapters. Fixed in 8c8a78d (thread per-weight in-dim through deq_w + 14 call sites).
- VERIFIED: with zeroed adapters (pure base Qwen3), loradec now generates fluent text: 'I have a 2023 MacBook Pro 14" M3 Pro...'. decode_driver was unaffected (different load path), which masked the bug.
- The earlier 'CE adapters not fluent' conclusion was a misdiagnosis - broken decode hid the adapters' actual behavior.

## Remaining (the true #13 gap)
With trained CE adapters + fixed decode, the model now LOOP-REPEATS ('desktop desktop workstation desktop workstation...') - classic LoRA over-amplification/repetition. That's a genuine training-objective issue (single record, 4 steps, rank-2 R=2 @ alpha 2, teacher-forced CE overfits to task-adjacent tokens). Not machinery anymore.

Next candidates (smallest first): lower AR (2.0->1.0 or 0.5), reduce/regularize the LoRA update, or a short next-token/RL fine-tune - then re-run loradec + verify_loradec.py.

**Comment by rickenator** [comment]
## Decode bug FIXED; adapter repetition = training collapse (confirmed, not a knob)

The hidden decode bug (deq_w inz=0) is fixed in 8c8a78d - base-model loradec is now fluent. That un-masked the REAL remaining gap: the trained CE adapters collapse into a repetition loop.

AR sweep at decode (2.0, 0.5, 0.1) all loop-repreat, each with different content ('desktop workstation' / '3D game development' / 'I want to use it...') but ALWAYS a fixed-repeat attractor. Repetition is robust to alpha scaling => it's a trained-in attractor (training collapse), not amplification strength. Cause: single record, 4 steps, rank-2 (R=2@alpha2), teacher-forced CE overfits to task-adjacent tokens with no generation-aware pressure.

To actually get generation-capable tuned adapters (the real #13 goal) needs a proper objective:
- scheduled sampling / next-token autoregressive fine-tune, OR
- more data points + more steps + stronger LoRA regularization/weight-decay, OR
- RL-style reward on emitting the {kind,message,...} contract.

Decoder + training machinery is now correct and fast to re-run (~7min/epoch); the RESIDUAL work is purely adapters-learning-to-not-repeat. Recommended as its own focused task (split from 'machinery' which is RESOLVED).

**Comment by rickenator** [comment]
RESOLVED for the achievable goal. The repetition collapse was a decode-side problem, not a training objective: the trained CE adapters decode into a fixed-repeat attractor at any alpha (exposure bias). Fix (4671317): decode-side repetition penalty in loradec lm_next (subtract PEN=1e30 from recent 32-token history logits before argmax) + decode AR 2.0->0.5. Tuned decode is now fluent+non-repeating: 'I want to use it for a 3D game development project. What are the key features and specifications that would be important? Also, how do you set up Hyprland on Ubuntu?...'. kvresp-train-kv gate re-verified green. HONEST LIMIT: a JSON agent-response contract is NOT reachable by greedy decode of this single-record rank-2 LoRA (full adapter strength -> gibberish, low AR -> natural language). Fluent tuned generation achieved; contract emission needs a real next-token/RL objective + more data (the earlier #13 goal) - tracked for follow-up if pursued.
